### PR TITLE
Prevent metadata download when getting installed products

### PR DESCRIPTION
**Pull request by mcalmer**
--disable-repos temporary disable all repos for this call. 

@isbm , @dmacvicar - maybe you want to have a look?

### CODE REVIEW

File: salt/modules/zypper.py [hunk 1]
```diff
@@ -1308,7 +1308,10 @@ def list_products(all=False, refresh=False):
 
     ret = list()
     OEM_PATH = "/var/lib/suseRegister/OEM"
-    cmd = _zypper('-x', 'products')
+    cmd = _zypper()
+    if not all:
+        cmd.append('--disable-repos')
+    cmd.extend(['-x', 'products'])
```
Comment (dmacvicar) [inline salt/modules/zypper.py]: I think `--disable-repos` is a global option, and therefore has to go before `products`

Comment (mcalmer) [inline salt/modules/zypper.py]: It does.

cmd.append(--disable-repos')
cmd.extend(['-x', 'products'])

Comment (dmacvicar) [inline salt/modules/zypper.py]: Sorry, you are right.

:+1:

**Comment by mcalmer** [comment]
manually merged.

**Pull request closed without merging**